### PR TITLE
Allow different buttons setup for tinyMCE

### PR DESCRIPTION
**Pull request by dgrammatiko**
Pull Request complimentary for Issue #14390 .

### Summary of Changes
Pass different params for each instance of tinyMCE

### Testing Instructions

- In the global configuration select Tynimce as your default editor.
- Create a new Editor custom field.

- Select the Show Buttons as YES!
- Select in the next input the buttons you want to hide.
- Save the field.
- Create a new article

- Repeat the process but making Show Buttons as NO!
 
### Expected result
Buttons behave according to the given input
screen shot 2017-03-07 at 22 57 58
screen shot 2017-03-07 at 22 58 19

### Documentation Changes Required
This might not be B/C compatible (depending the tinyMCE introduced the Joomla.storageOptions)

### CODE REVIEW

**Comment by zero-24** [comment]
> This might not be B/C compatible (depending the tinyMCE introduced the Joomla.storageOptions)

As this was done in 3.7 this is not a problem right?

**Comment by dgt41** [comment]
I couldn't find the PR for that, thanks for clearing this out

**Comment by zero-24** [comment]
should be this here: https://github.com/joomla/joomla-cms/pull/11157 @dgt41

**Comment by dgt41** [comment]
@Fedik can you review the changes here?

**Comment by laoneo** [comment]
Is this ready for testing?

**Comment by dgt41** [comment]
Yup

**Comment by laoneo** [comment]
I have tested this item :white_check_mark: successfully on d8d404c719cd18c15b178b79f256c53f11cf2c15<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/14417">issues.joomla.org/tracker/joomla-cms/14417</a>.</sub>

**Comment by infograf768** [comment]
I have tested this item :white_check_mark: successfully on d8d404c719cd18c15b178b79f256c53f11cf2c15<br /><br />Works nicely.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/14417">issues.joomla.org/tracker/joomla-cms/14417</a>.</sub>

**Comment by infograf768** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/14417">issues.joomla.org/tracker/joomla-cms/14417</a>.</sub>

File: plugins/editors/tinymce/tinymce.php [hunk 1]
```diff
@@ -701,7 +701,7 @@ public function onDisplay($name, $content, $width, $height, $col, $row, $buttons
 		$options['tinyMCE']['default'] = $scriptOptions;
 
 		$doc->addStyleDeclaration('.mce-in { padding: 5px 10px !important;}');
-		$doc->addScriptOptions('plg_editor_tinymce', $options);
+		$doc->addScriptOptions('plg_editor_tinymce_' . $id, $options);
```
Comment (Fedik) [inline plugins/editors/tinymce/tinymce.php]: I am afraid this is not compatible with subform,
There the id will be generated by JS, therefore the new editor instances (added in new row by JS) will not have any defined options while call `Joomla.getOptions('plg_editor_tinymce_' + editor.id, {})`

**Comment by Fedik** [comment]
@dgt41 I am afraid it will fail in multiple cases
but, there is possibility to override default options for the editor by the editor field name, introduced in #11157, maybe we can use it somehow here. Also 'default'  options are cached.

**Comment by infograf768** [comment]
shall rtc be removed?

**Comment by dgt41** [comment]
yup, let's figure out another way for this

**Comment by infograf768** [comment]
back to pending<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/14417">issues.joomla.org/tracker/joomla-cms/14417</a>.</sub>

**Comment by coolcat-creations** [comment]
Would be great if it will be possible, i ran into this today :) 👍

**Comment by dgt41** [comment]
@Fedik how about: https://github.com/joomla/joomla-cms/pull/14417/commits/5d1b0d995fd96582116a5813884f0ece31ae0f65

**Comment by Fedik** [comment]
@dgt41 could work, but still not perfect 😉 
did you seen, there already test for "whether default parameters loaded"? 😉  https://github.com/dgt41/joomla-cms/blob/5d1b0d995fd96582116a5813884f0ece31ae0f65/plugins/editors/tinymce/tinymce.php#L203

I think here need something like:
```javascript
// Load default:
if (empty($options['tinyMCE']['default'])) {
  $options['tinyMCE']['default'] = array(/* default, general, params*/)
}

// Load changes for current name
$options['tinyMCE']['fieldName']['toolbar1'] = 'blabla';
```

**Comment by dgt41** [comment]
@Fedik I know, but the other option needs a lot of refactoring and due to the legacy method is double work. We can postpone it for J4 when we will do the clean up.

**Comment by Fedik** [comment]
> I know

then you should note that adding `if (!$this->defaultParamsLoaded)` make no difference 😉 

> but the other option needs a lot of refactoring and due to the legacy method is double work. 

true, but current solution with `_' . $id` introduce more mess , I think

I try to check more, this weekend, if will get some time 😉

**Comment by Fedik** [comment]
@dgt41 check this #14520

**Comment by dgt41** [comment]
there is a new PR #14520

**Pull request closed without merging**